boot/build.go

```diff
@@ -281,6 +291,43 @@ func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
 	return result, nil
 }
 
+func lookForServicesFileSystemProviderAndRemoveNested(context libcnb.BuildContext) error {
+	dir := filepath.Join(context.Application.Path, "META-INF", "services")
+	if servicesFolderExists, err := sherpa.DirExists(dir); err != nil {
+		return fmt.Errorf("unable to check directory %s\n%w", dir, err)
+	} else if servicesFolderExists {
+		fileSystemProvider := filepath.Join(dir, FileSystemProvider)
+
+		content, err := os.ReadFile(fileSystemProvider)
+		if os.IsNotExist(err) {
+			return nil
+		} else if err != nil {
+			return fmt.Errorf("unable to read %s\n%w", fileSystemProvider, err)
+		}
+		originalLines := strings.Split(string(content), "\n")
+		var newLines []string
+		for _, line := range originalLines {
+			if !strings.Contains(line, NestedFileSystemProvider) {
+				newLines = append(newLines, line)
+			}
+		}
+
+		if len(newLines) > 0 {
+			err = os.WriteFile(fileSystemProvider, []byte(strings.Join(newLines, "\n")), 0644)
+			if err != nil {
+				return fmt.Errorf("unable to write to %s\n%w", fileSystemProvider, err)
+			}
+		} else {
+			err = os.Remove(fileSystemProvider)
+			if err != nil {
+				return fmt.Errorf("unable to delete file %s\n%w", fileSystemProvider, err)
+			}
```

Comment:
This really isn't a big deal because this is not a performance-critical section, but you can avoid converting to strings here since the parsing/manipulation you're doing is simple. The `bytes` package has similar methods for `Split`, `Contains` and `Join`, so you could use those instead and operate directly on the bytes.

https://pkg.go.dev/bytes

Reply:
okee dok! done! thanks!